codegen/src/main/java/software/amazon/awssdk/codegen/customization/processors/EventStreamSharedEventProcessor.java

```diff
@@ -0,0 +1,143 @@
+                    throw new IllegalStateException(
+                        String.format("Cannot find event member [%s] in the eventstream [%s] when processing "
+                                      + "customization config duplicateAndRenameSharedEvents.%s",
+                                      eventEntry.getKey(), eventStreamName, eventStreamName));
+                }
+
+                String shapeToDuplicate = eventMemberToModify.getShape();
+                Shape eventMemberShape = serviceModel.getShape(shapeToDuplicate);
+
+                if (eventMemberShape == null || !eventMemberShape.isEvent()) {
+                    throw new IllegalStateException(
+                        String.format("Error: [%s] must be an Event shape when processing "
+                                      + "customization config duplicateAndRenameSharedEvents.%s",
+                                      eventEntry.getKey(), eventStreamName));
+                }
+
+                String newShapeName = eventEntry.getValue();
+                if (serviceModel.getShapes().containsKey(newShapeName)) {
+                    throw new IllegalStateException(
+                        String.format("Error: [%s] is already in the model when processing "
+                                      + "customization config duplicateAndRenameSharedEvents.%s",
+                                      newShapeName, eventStreamName));
+                }
+                serviceModel.getShapes().put(newShapeName, duplicateShape(eventMemberShape));
+                eventMemberToModify.setShape(newShapeName);
+                log.info("Duplicated and renamed event member on {} from {} -> {}",
+                         eventStreamName, shapeToDuplicate, newShapeName);
+            }
+        }
+    }
+
+    private Shape duplicateShape(Shape shape) {
+        StringWriter writer = new StringWriter();
+        try {
+            Jackson.writeWithObjectMapper(shape, writer);
+            Shape duplicated = Jackson.load(Shape.class, writer.toString());
+            duplicated.setSynthetic(true);
+            return duplicated;
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static void validateIsEventStream(Shape shape, String name) {
+        if (shape == null) {
+            throw new IllegalStateException(
+                String.format("Cannot find eventstream shape [%s] in the model when processing "
+                              + "customization config duplicateAndRenameSharedEvents.%s", name, name));
+        }
+        if (!shape.isEventstream()) {
+            throw new IllegalStateException(
+                String.format("Error: [%s] must be an EventStream when processing "
+                              + "customization config duplicateAndRenameSharedEvents.%s", name, name));
+        }
+    }
+
+    @Override
+    public void postprocess(IntermediateModel intermediateModel) {
```

Comment:
Can we add a test case to cover the post processing codepath?

Reply:
There is a test in `EventStreamSharedEventProcessorTest` - see: `modelWithSharedEvents_raises` - the test doesn't call postprocess directly, but instead relies on the IntermediateModel/C2jModel builders to call all of the processors.